Bmx.Idp.Okta/OktaClient.cs

```diff
@@ -46,37 +58,62 @@ public class OktaClient : IIdentityProvider {
 		}
 
 		public async Task<bool> ChallengeMfa( int selectedMfaIndex, string challengeResponse ) {
-			var stateToken = ( (AuthenticateResponseInital)_authenticateResponses[AuthenticationStatus.MFA_REQUIRED] )
-				.StateToken;
-			var mfaFactor =
-				( (AuthenticateResponseInital)_authenticateResponses[AuthenticationStatus.MFA_REQUIRED] ).Embedded
-				.Factors[selectedMfaIndex];
+			var mfaFactor = _oktaMfaFactors[selectedMfaIndex];
 
 			var authResp =
 				await AuthenticateChallengeMfaOkta(
-					new AuthenticateChallengeMfaOptions( mfaFactor.Id, challengeResponse, stateToken ) );
+					new AuthenticateChallengeMfaOptions( mfaFactor.Id, challengeResponse, _oktaStateToken ) );
 
-			_authenticateResponses.Add( AuthenticationStatus.SUCCESS, authResp );
+			_oktaSessionToken = authResp.SessionToken;
 
 			// TODO: Check for auth successes and return state
 			return true;
-			// TODO: Implicitly Call Session here
+		}
+
+		public async Task<string[]> GetAccounts( string accountType ) {
+			// TODO: Use existing session if it exists in ~/.bmx and isn't expired
+			var sessionResp = await CreateSessionOkta( new SessionOptions( _oktaSessionToken ) );
+			AddSession( sessionResp.Id );
+
+			_oktaApps = await GetAccountsOkta( sessionResp.UserId );
+			return _oktaApps.Where( app => app.AppName == accountType ).Select( app => app.Label ).ToArray();
+		}
+
+		private void AddSession( string sessionId ) {
+			CookieContainer.Add( new Cookie( "sid", sessionId, "/", HttpClient.BaseAddress.Host ) );
 		}
 
 		// TODO: Consider consolidating this kind of thing, ex: to a OktaHttpClient
 		private async Task<AuthenticateResponseInital> AuthenticateOkta( AuthenticateOptions authOptions ) {
 			var resp = await HttpClient.PostAsync( "authn",
-				new StringContent( JsonSerializer.Serialize( authOptions, _serializeOptions ) ) );
+				new StringContent( JsonSerializer.Serialize( authOptions, _serializeOptions ), Encoding.Default,
+					"application/json" ) );
+
 			return await JsonSerializer.DeserializeAsync<AuthenticateResponseInital>(
 				await resp.Content.ReadAsStreamAsync(), _serializeOptions );
 		}
 
 		private async Task<AuthenticateResponseSuccess> AuthenticateChallengeMfaOkta(
 			AuthenticateChallengeMfaOptions authOptions ) {
 			var resp = await HttpClient.PostAsync( $"authn/factors/{authOptions.FactorId}/verify",
-				new StringContent( JsonSerializer.Serialize( authOptions, _serializeOptions ) ) );
+				new StringContent( JsonSerializer.Serialize( authOptions, _serializeOptions ), Encoding.Default,
+					"application/json" ) );
 			return await JsonSerializer.DeserializeAsync<AuthenticateResponseSuccess>(
 				await resp.Content.ReadAsStreamAsync(), _serializeOptions );
 		}
+
+		private async Task<OktaSession> CreateSessionOkta( SessionOptions sessionOptions ) {
+			var resp = await HttpClient.PostAsync( "sessions",
```

Comment:
Similar: `HttpClientExtensions` has a `PostAsJsonAsync()` in case you haven't considered it already.